scripts/doc_checker.py

```diff
@@ -307,8 +308,15 @@ def get_noqa_checks(doc: Docstring) -> list:
                 noqa_str = line
                 break
     else:
-        # pydocstyle only check first line; aling with it
-        noqa_str = source.split("\n", 1)[0]
+        # noqa string is defined as the first line before the docstring
+        if not doc.raw_doc:
+            # noqa string is meaningless if there is no docstring in module
+            return []
+        lines = source.split("\n")
+        for idx, line in enumerate(lines):
+            if '"""' in line or "'''" in line:
+                noqa_str = lines[idx - 1]
```

Comment:
you risk taking _last line_ of file if multiline docstring happens to start at the beginning of the file :)

Reply:
Agree, but I'm relying on the main use case for docstring usage in Modin modules.